src/wallet/wallet.cpp

```diff
@@ -146,6 +146,15 @@ std::shared_ptr<CWallet> LoadWallet(interfaces::Chain& chain, const WalletLocati
     }
     AddWallet(wallet);
     wallet->postInitProcess();
```

Comment:
Any reason not to do this in `postInitProcess`?

I understand that this is also called when the `createwallet` rpc is called, but the post-init seems useless in that case anyway.

I'd suggest:
* Remove `postInitProcess` from `createwallet`, because the wallet is fresh and there couldn't be any txs that need to be exchanged between mempool and wallet
* Rename `postInitProcess` to `postLoadProcess`, since it is not needed on init, but on load
* Move the below code into `postLoadProcess` with mempool::cs hold while adding the txs.

Reply:
I agree, `postInitProcess` is "pre dynamic wallet support". However I'd prefer to refactor in a different PR, this is a bug fix.

Reply:
Fine with doing the rename et al. later, but is there any reason not to put it in `postInitProcess`? IIRC it was in there previously in this pull

Reply:
Yes, could be moved back there.